src/journal-remote/journal-remote-util.c

```diff
@@ -0,0 +1,89 @@
+#include "extract-word.h"
+#include "journal-remote-util.h"
+#include "parse-util.h"
+
+void compression_args_clear(CompressionArgs *args) {
+        args->size = 0;
+        free(args->opts);
+}
+
+int config_parse_compression(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        CompressionArgs *args = ASSERT_PTR(data);
+        int r;
+        bool parse_level = ltype;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+
+        if (isempty(rvalue)) {
+                compression_args_clear(args);
+                return 1;
+        }
+
+        for (const char *p = rvalue;;) {
+                _cleanup_free_ char *algorithm = NULL, *word = NULL;
+                int level = -1;
+
+                r = extract_first_word(&p, &word, NULL, 0);
+                if (r < 0)
+                        return log_syntax_parse_error(unit, filename, line, r, lvalue, rvalue);
+                if (r == 0)
+                        break;
+
+                const char *q = word;
+                r = extract_first_word(&q, &algorithm, ":", 0);
```

Comment:
Even `parse_level` is false, still level specifier is supported, e.g. `xz:5`. Please refuse such and only accept algorithm, e.g. `xz`.